### ISSUE TITLE
Consider moving from Centos7 to RHEL8 for 18.x

### ISSUE THREAD

**Issue by AshCripps** [nodejs/build#2741]
Whilst looking at https://github.com/nodejs/build/issues/2740 ive discovered that Centos7 goes EOL by the end of 18.x, so we should consider moving off of centos7 before 18.x is cut next year.

A suitable option would be to upgrade to RHEL8 - Red Hat is providing a scheme for open source communities to gain free access to RHEL subscriptions but wanted to pose the question to the wider group to see if anyone had any objections?

cc @nodejs/build

**Comment by richardlau** [comment]
@AshCripps I don’t think we have enough ppc Linux hosts for both CentOS7 (existing workloads) and the next (e.g. RHEL8).

**Comment by AshCripps** [comment]
we are nowhere near our capacity for ppc linux machines atm (provided we are allowed to use everything given to us)

**Comment by mhdawson** [comment]
Confirming @AshCripps assessment that we should have no problem adding more ppc machines. We should look at the same for other architectures as well.

Either way I think it's something we have to figure out as I don't think we want to be building on an EOL Operating system.  One of the main concerns is that it will main an increase in the minimum glibc levels...

**Comment by richardlau** [comment]
One thing we'd need to check for the x64 platforms is what is offered on the various cloud providers and if RHEL 8 is an out of the box offering (i.e. existing selectable image).

**Comment by richardlau** [comment]
| Provider | RHEL 8 image available |
|---|---|
| DigitalOcean | No |

# DigitalOcean
![image](https://user-images.githubusercontent.com/5445507/131831558-7b80f1fc-e499-407a-b84b-fe75d3353b9e.png)

**Comment by richardlau** [comment]
So it looks like currently only IBM offer an OOTB RHEL image (of any version). On DigitalOcean we can use [custom images](https://docs.digitalocean.com/products/images/custom-images/) but that's going to be some work to set up. Or we run RHEL in a container via Docker under a different host OS.

**Comment by AshCripps** [comment]
I assume all the other OS are free version? RHEL probably doesnt show up as it requires a subscription

**Comment by richardlau** [comment]
I've started adding RHEL8 s390x (LinuxONE) in https://github.com/nodejs/build/pull/2859.

I've also requested a no-cost RHEL subscription under the [Red Hat for Open Source Infrastructure](https://www.redhat.com/en/blog/extending-no-cost-red-hat-enterprise-linux-open-source-organizations) program. We've had a positive response and this is in progress -- I was asked to create a Red Hat account to attach the entitlements to and I have created the account (details in the secrets repo) and am waiting for the entitlements.

**Comment by richardlau** [comment]
As a follow up, we do have a RHEL subscription now -- details are in the secrets repository. 

The LinuxONE machines have been added and running now with nightly binaries for master now being built on RHEL 8.

I've added (well mainly repurposed some obsolete CentOS 6 instances) some x64 RHEL8 instances at IBM Cloud -- the plan is to add/repurpose more at Digital Ocean/RackSpace but there is more work to be done there as we'll need to either roll our own images or potentially containerize.

**Comment by richardlau** [comment]
We now have RHEL 8 machines at the following providers:
- Digital Ocean, x64. I've uploaded a custom image [1].
- IBM Cloud, x64.
- OSUOSL, Linux ppc64le. I've uploaded a custom image [1].

[1] Red Hat provide a QCOW2 format "Red Hat Enterprise Linux 8.5 Update KVM Guest Image" which Digital Ocean and the OpenStack instance at OSUOSL accept as a custom image that we can use to provision.

RackSpace needs a bit more work -- we'd need to convert the image into another format.

The remaining CentOS 7 in the release CI is for arm64. I've had discussions with Equinix and have sent an email re. the current situation about getting an additional arm64 release machine to the build email address.
